resources/tentative/todomvc/src/components/todo-app/todo-app.component.js

```diff
@@ -0,0 +1,162 @@
+import template from "./todo-app.template.js";
+import TodoTopbar from "../todo-topbar/todo-topbar.component.js";
+import TodoList from "../todo-list/todo-list.component.js";
+import TodoBottombar from "../todo-bottombar/todo-bottombar.component.js";
+import { useRouter } from "../../hooks/useRouter.js";
+class TodoApp extends HTMLElement {
+    constructor() {
+        super();
+        // state
+        this._isReady = false;
+        this._data = [];
+        // elements
+        const node = document.importNode(template.content, true);
+        this.topbar = new TodoTopbar();
+        this.list = new TodoList();
+        this.bottombar = new TodoBottombar();
+        node.querySelector("[name=\"topbar\"]").append(this.topbar);
```

Comment:
I think it's rather unusual for component to be instantiating each subcomponent manually like this instead of relying on custom elements to create them automatically.

Reply:
@rniwa @rictic - seemed like an easy change, but threw me off for a little bit. 
I made the change, adding the custom elements directly in the template file. 

The order of the scripts in the head of the html also matter, since I need to ensure those used in the template file are added first, so they can register properly before the app tries to interact with them.

Let me know what you think.

Reply:
Looks good to me.